RestaurantReservation.API/Controllers/ReservationsController.cs

```diff
@@ -0,0 +1,208 @@
+        /// <response code="404">If the restaurant or table specified in the reservation does not match</response>
+        [HttpPost]
+        public async Task<ActionResult<MenuItemDto>> Create(ReservationDto reservationDto)
+        {
+            var matchingRestaurantTable = await _reservationRepository.MatchingTableRestaurant(reservationDto.RestaurantId,reservationDto.TableId);
+            if (!matchingRestaurantTable )
+            {
+                var error = ReservationErrors.NotMatchingTableRestaurant(reservationDto.RestaurantId, reservationDto.TableId);
+                return NotFound(error);
+            }
+            var reservationToAdd = _mapper.Map<Reservation>(reservationDto);
+            await _reservationRepository.AddAsync(reservationToAdd);
+            var reservationToReturn = _mapper.Map<ReservationWithoutDetailsDto>(reservationToAdd);
+            return CreatedAtRoute("GetReservation", new { id = reservationToReturn.Id }, reservationToReturn);
+        }
+
+        /// <summary>
+        /// Delete reservation by Id
+        /// </summary>
+        /// <param name="id">Id of reservation to delete</param>
+        /// <returns>No content if success</returns>
+        /// <response code="204">If reservation is deleted successfully</response>
+        /// <response code="404">If no reservation with the specified id is found</response>
+        [HttpDelete("{id}")]
+        public async Task<ActionResult> Delete(int id)
+        {
+            var existingReservation = await _reservationRepository.GetByIdAsync(id);
+            if (existingReservation == null)
+            {
+                var error = ReservationErrors.NotFound(id);
+                return NotFound(error);
+            }
+            await _reservationRepository.DeleteAsync(existingReservation);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Update reservation by Id
+        /// </summary>
+        /// <param name="id">Id of reservation to update</param>
+        /// <param name="reservation">The updated reservation data</param>
+        /// <returns>No content if success</returns>
+        /// <response code="204">If reservation is updated successfully</response>
+        /// <response code="404">If no reservation with the specified id is found or the restaurant or table specified in the reservation does not match </response>
+        [HttpPut("{id}")]
+        public async Task<ActionResult> Update(int id, ReservationDto reservation)
+        {
+            var existingReservation = await _reservationRepository.GetByIdAsync(id);
+            if (existingReservation == null)
+            {
+                var error = ReservationErrors.NotFound(id);
+                return NotFound(error);
+            }
+
+            var matchingRestaurantTable = await _reservationRepository.MatchingTableRestaurant(reservation.RestaurantId, reservation.TableId);
```

Comment:
mmm, maybe we can just add this to a validation function where we make sure that this table related to this restuarant. but I would suggest also to register to the table directly without passing the restaurnt so we won't need to validate this. - would explain.